### ISSUE TITLE
`cardano-cli transaction view` deserialise error

### ISSUE THREAD

**Issue by cblp** [input-output-hk/cardano-node#3224]
Passing a signed tx looks great, but passing a txbody throws a deserialise error:

```
cardano-cli transaction view --tx-body-file shelley/tx-proposal-d-0.txbody
Command failed: transaction view  Error: shelley/tx-proposal-d-0.txbody: TextEnvelope decode error: DecoderErrorDeserialiseFailure "Shelley TxBody" (DeserialiseFailure 1 "expected tx body tuple of size 3, 4 or 6")
```

_Originally posted by @disassembler in https://github.com/input-output-hk/cardano-node/issues/3208#issuecomment-921069152_

**Comment by cblp** [comment]
@disassembler please show how you prepared the txbody

**Comment by dcoutts** [comment]
Lets clarify what problem we are solving here. As I understand it:

In the code we had at the time of the Shelley hard fork but before the Alonzo hard fork, the CLI would serialise tx bodies as a 2-tuple. So txbody files saved at that time would use that 2-tuple format.

We've not had backwards compat support for that 2-tuple format ever since we added multi-sig script support to the API & CLI, which was nearly a year ago. 

So issue is about _restoring_ support for an old format that we have not supported for the last ~9 months or so. Apparently we've had only a single (internal) request to restore this support.

**Comment by dcoutts** [comment]
I stand corrected: the change was not a year ago but in 1.27.0 in May this year.
